api/app/tests/apps/signals/test_utils.py

```diff
@@ -100,3 +100,8 @@ def test_non_existing_formats(self):
 
         formatted_address_str = address_formatter.format(format_str='Z')
         self.assertEqual(formatted_address_str, 'Z')
+
+    def test_address_none_type(self):
+        address_formatter = AddressFormatter(address=None)
+        formatted_address_str = address_formatter.format(format_str='O')
+        self.assertEqual(formatted_address_str, '')
```

Comment:
You should test the other formatting options as well (not just `'O'`).

Reply:
Added them